python/pyspark/pandas/indexes/base.py

```diff
@@ -2292,9 +2292,7 @@ def union(
 
         sdf_self = self._internal.spark_frame.select(self._internal.index_spark_columns)
         sdf_other = other_idx._internal.spark_frame.select(other_idx._internal.index_spark_columns)
-        sdf = sdf_self.union(sdf_other.subtract(sdf_self))
-        if isinstance(self, MultiIndex):
-            sdf = sdf.drop_duplicates()
+        sdf = sdf_self.unionAll(sdf_other).exceptAll(sdf_self.intersectAll(sdf_other))
         if sort:
```

Comment:
@itholic, did you refer to this?

> Bug in Index.union() and MultiIndex.union() dropping duplicate Index values when Index was not monotonic or sort was set to False (GH36289, GH31326, GH40862)

It sounds like now it drops duplicates but the codes above look keeping the duplicates. Should we also at least check `sort` and drop duplicates?

Reply:
Thanks, @HyukjinKwon !

Seems like they said "dropping duplicate Index values" was a bug, and fixed in this release.

Related discussion was made at https://github.com/pandas-dev/pandas/issues/36289.

Also checked the `sort` and drop duplicates, works the same as pandas.